### ISSUE TITLE
error[E0277]: the trait bound `num_complex::Complex<f64>: ndarray:: ScalarOperand` is not satisfied

### ISSUE THREAD

**Issue by macthecadillac** [rust-ndarray/ndarray#829]
I find this to be extremely odd but I couldn't wrap my head around the error I'm getting. Am I doing something stupid, or is it genuinely a bug?

num-complex version: 0.3
ndarray version: 0.13

```rust
extern crate num_complex;
extern crate ndarray;

use num_complex::Complex64;
use ndarray::prelude::*;

fn main() {
    let ar = array![Complex64::new(1., 9.), Complex64::new(2., 0.)];
    println!("{}", ar * Complex64::new(1., 2.));
}
```

Cargo output:
```
error[E0277]: the trait bound `num_complex::Complex<f64>: ndarray::ScalarOperand` is not satisfied
 --> src/main.rs:9:23
  |
9 |     println!("{}", ar * Complex64::new(1., 2.));
  |                       ^ the trait `ndarray::ScalarOperand` is not implemented for `num_complex::Complex<f64>`
  |
  = note: required because of the requirements on the impl of `std::ops::Mul<num_complex::Complex<f64>>` for `ndarray::ArrayBase<ndarray::OwnedRepr<num_complex::Complex<f64>>,ndarray::Dim<[usize; 1]>>
`
```

**Comment by SuperFluffy** [comment]
Your code actually works: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=e5fc28f763c1474edee39d98f2a21aa9

I trip over this kind of issue myself from time to time. What you have discovered is probably some mismatch in libraries. Rust is not doing a very good job right now at telling you what exactly the issue is, and you sometimes stumble upon absurd situations where you get errors akin to `trait Foo is not implemented for Bar`, even though you *know* that your object is indeed a `Bar`, and rust further seems to be making fun of you by suggesting that you want to use `Bar`, because that one actually has `Foo`, you know?! :-D

The issue is that your local `bar: Bar` comes from `somecrate = v0.2.0`, while `Foo` is implemented for `Bar` from `somecrate = v0.1.0`.

Having written all that down, I suggest you do 2 things:

1. Run `cargo tree` and check the version of `num_complex` that you are using, and also which version of `ndarray` you really have and which `num_complex` `ndarray` pulls in.
2. Run `cargo update` followed by `cargo build`.

**Comment by macthecadillac** [comment]
Thanks. It worked after I matched the num-complex version with the one that got pulled in by ndarray.
